### DISCUSSION TITLE
How to integrate my data to do scRNA analysis

### DISCUSSION THREAD

**Discussion by zandigohar** [satijalab/seurat#4197] [Q&A] [answered]
I want to do scRNA analysis but I am confused how to merge/integrate the data. There are 3 different conditions present: healthy, disease type 1, and disease type 2.

Some of my scRNA data has one _count matrix_ (.mtx), _features_, and _barcodes_ files each, but the rest have double _count matrix_ (.mtx), _features_, and _barcodes_ files each because they were previously run into 2 lanes (L1 and L2) and the files were created separately. So I want to kinda merge those that are run into two lanes into one. (I understand there is a procedure in Cellranger to make that happen but I do not have the fastq/etc files but only those 3 that I mentioned earlier, so that's off the table)

I read these samples with `Read10X()` and then created each object by `CreateSeuratObject()` function. Then I believe I have to merge those objects that have L1 and L2.

Now, I would use the data integration standard procedure available in the corresponding Seurat v4 vignette:

```
features <- SelectIntegrationFeatures(object.list = obj.list)
immune.anchors <- FindIntegrationAnchors(object.list = obj.list, anchor.features = features)
immune.combined <- IntegrateData(anchorset = immune.anchors)
```

Where `obj.list` is the list of those Seurat objects while L1 and L2 objects are merged.

So, my question is:
1. Is there a better option than merge() to combined those objects that are L1 and L2 of the same sample? Does merging make sense at all?
2. I can also make 3 objects based on those 3 conditions and then do the data integration using `merge()` and then `SelectIntegrationFeatures()` and so on, or, do I do data integration for the 13 Seurat objects without merging?
3. I did NormalizeData and FindVariableFeatures for every 16 objects first and after data integration, I do scaling. Is that the correct approach?

**Comment by timoast** [comment, marked as answer]
> Some of my scRNA data has one count matrix (.mtx), features, and barcodes files each, but the rest have double count matrix (.mtx), features, and barcodes files each because they were previously run into 2 lanes (L1 and L2) and the files were created separately.

You unfortunately can't just merge the count matrices derived from different sequencing runs of the same library, as the result would be incorrectly quantified (double counting of UMIs). You need to fix that issue by running `cellranger aggr` before doing anything else. If you really don't have access to the original data, then you need to drop one of the sequencing runs.

In general, we recommend that you first combine datasets by merging them rather than performing integration, and check whether there is a batch effect that needs to be corrected. If so, you should then use the data integration tools in Seurat.
